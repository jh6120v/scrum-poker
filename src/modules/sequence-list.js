import { createActionCreator } from '../commons/utils';
import { createReducer } from '../store/reducers';

// Actions
const actionCreator = createActionCreator('@@POKER/SEQUENCE_LIST');
const sequenceListDataFetch = actionCreator('DATA_FETCH');
const sequenceListDataSet = actionCreator('DATA_SET');

export { sequenceListDataFetch, sequenceListDataSet };


// Reducers
const initialState = {
    sequenceList: {},
    cache: false
};

const handlers = {
    [sequenceListDataSet.type]: (state, { payload: { sequenceList } }) => ({
        ...state,
        sequenceList,
        cache: true
    })
};

const reducers = createReducer(initialState, handlers);
export default reducers;
