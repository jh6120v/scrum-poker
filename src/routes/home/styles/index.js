import styled, { keyframes } from 'styled-components';
import { respondTo } from '../../../styles/_mixin';

const Container = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  align-content: center;
  height: 100%;
  padding: 30px 15px;
  background: #e2e2e2;
  background: -moz-linear-gradient(45deg,  #e2e2e2 0%, #dbdbdb 50%, #d1d1d1 51%, #fefefe 100%);
  background: -webkit-linear-gradient(45deg,  #e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  background: linear-gradient(45deg,  #e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#e2e2e2', endColorstr='#fefefe',GradientType=1 );
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Logo = styled.div`
  width: calc((100% - 100px) / 2);
  height: 200px;
  background: url(${(props) => props.img}) no-repeat center;
  background-size: contain;
  animation: ${rotate} 3s linear infinite;
  ${respondTo.sm`
    width: 220px;
  `}
`;

const Title = styled.h1`
  width: 100%;
  font-size: 40px;
  text-align: center;
  padding: 30px 0;
`;

const backAndForth = keyframes`
  0% {
    left: 0;
  }
  50% {
    left: calc(100% - 2rem);
    background-color: #8660C2;
  }
  100% {
    left: 0;
  }
`;

const LogoLink = styled.div`
  width: 80px;
  margin: 0 10px;
  position: relative;
  &:before {
    content: '';
    width: 1rem;
    height: 1rem;
    background-color: #62DAFB;
    border-radius: 50%;
    position: absolute;
    left: 0;
    top: calc(50% - 0.5rem);
    transition: all 1s ease-in-out;
    animation: ${backAndForth} 1.7s infinite;
  }
  ${respondTo.xs`
    &:before {
      width: 2rem;
      height: 2rem;
      top: calc(50% - 1rem);
    }
  `}
`;

export {
    Container,
    Logo,
    Title,
    LogoLink
};
