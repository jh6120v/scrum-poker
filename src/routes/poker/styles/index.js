import styled, { css } from 'styled-components';
import { FlipCard, FlipCardFront, FlipCardBack } from './flip-card';
import { respondTo } from '../../../styles/_mixin';
import logo from '../../../assets/images/bugcat-logo-bg.png';
import logo2 from '../../../assets/images/bugcat-logo-text-bg.png';
import cardBg1 from '../../../assets/images/card-bg/card-1.gif';
import cardBg2 from '../../../assets/images/card-bg/card-2.gif';
import cardBg3 from '../../../assets/images/card-bg/card-3.gif';
import cardBg4 from '../../../assets/images/card-bg/card-4.gif';
import cardBg5 from '../../../assets/images/card-bg/card-5.gif';
import cardBg6 from '../../../assets/images/card-bg/card-6.gif';
import cardBg7 from '../../../assets/images/card-bg/card-7.gif';
import cardBg8 from '../../../assets/images/card-bg/card-8.gif';
import cardBg9 from '../../../assets/images/card-bg/card-9.gif';
import cardBg10 from '../../../assets/images/card-bg/card-10.gif';
import cardBg11 from '../../../assets/images/card-bg/card-11.gif';
import cardBg12 from '../../../assets/images/card-bg/card-12.gif';
import cardBg13 from '../../../assets/images/card-bg/card-13.gif';
import cofeeCat from '../../../assets/images/coffee-cat.png';

// flip
const PokerContainer = styled(FlipCard)`
  //height: calc(100vh - 135px);
  ${respondTo.sm`
    width: 350px;
    margin: 0 auto;
  `}
`;

const PokerListWrap = styled(FlipCardFront)`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-content: center;
  background: #aadff0;
  z-index: 1;
`;

const PokerSelectWrap = styled(FlipCardBack)`
  display: flex;
  justify-content: center;
  align-items: center;
  background: #aadff0;
  z-index: 1;
`;

// item auto scale. real poker radio
const PokerItemWrap = styled.div`
  display: flex;
  width: 100px;
  max-width: 22%;
  margin: 6.5px 0;
  &::after {
    content: "";
    display: block;
    padding-bottom: 140%; // 高度
  }
`;

const PokerItem = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.8rem;
  border-radius: 10px;
  background: #fff;
  cursor: pointer;
  overflow: hidden;
  position: relative;
`;

const PokerItemMask = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  transition: all 200ms linear;
  z-index: 0;
  ${(props) => {
        if (props.active) {
            return css`
              background: rgba(0,0,0,1);
            `;
        }
        return css`
          background: rgba(0,0,0,0);
        `;
    }};
`;

const PokerItemInner = styled.div`
  width: 90%;
  height: 94%;
  border-radius: 10px;
  color: #fff;
  background-color: #5bb1d4;
  display: flex;
  justify-content: center;
  align-items: center;
  text-shadow: 1px 1px 2px rgba(29, 29, 31, .7);
  background-image: url(${logo});
  background-repeat: no-repeat;
  background-position: center center;
  background-size: cover;
`;

const PokerSelectSafari = styled.div`
  width: 100%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  border-radius: 25px;
`;

const PokerSelectMask = styled(PokerItemMask)`
  z-index: 2;
  border-radius: 25px;
`;

const PokerSelectContainer = styled(FlipCard)`
  display: flex;
  width: 100%;
  height: auto;
  &:after {
     content: "";
     display: block;
     padding-bottom: 154%; // 高度
   }
`;

const PokerSelectedItemFront = styled(FlipCardFront)`
  display: flex;
  justify-content: center;
  border-radius: 25px;
  cursor: pointer;
  font-size: 5rem;
  color: #5AC9E8;
  background-color: #fff;
  background-size: contain;
  background-position: bottom;
  background-repeat: no-repeat;
  z-index: 1;
  &.img-1 {
    background-image: url(${cardBg1});
  }
  &.img-2 {
    background-image: url(${cardBg2});
  }
  &.img-3 {
    background-image: url(${cardBg3});
  }
  &.img-4 {
    background-image: url(${cardBg4});
  }
  &.img-5 {
    background-image: url(${cardBg5});
  }
  &.img-6 {
    background-image: url(${cardBg6});
  }
  &.img-7 {
    background-image: url(${cardBg7});
  }
  &.img-8 {
    background-image: url(${cardBg8});
  }
  &.img-9 {
    background-image: url(${cardBg9});
  }
  &.img-10 {
    background-image: url(${cardBg10});
  }
  &.img-11 {
    background-image: url(${cardBg11});
  }
  &.img-12 {
    background-image: url(${cardBg12});
  }
  &.img-13 {
    background-image: url(${cardBg13});
  }
`;

const PokerSelectedItemFrontWrap = styled.div`
  width: 100%;
  height: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  padding-top: 17%;
`;

const PokerSelectedItemFrontMask = styled(PokerItemMask)`
  z-index: 0;
  border-radius: 25px;
`;

const PokerSelectedItemBack = styled(FlipCardBack)`
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 25px;
  cursor: pointer;
  transform: rotateY(180deg);
  background-color: #fff;
  z-index: 1;
`;

const PokerSelectedItemBackWrap = styled(PokerSelectedItemFrontWrap)`
  align-items: center;
  padding-top: 0;
`;

const PokerSelectedItemBackMask = styled(PokerItemMask)`
  z-index: 2;
  border-radius: 25px;
`;

const PokerSelectedItemBackInner = styled(PokerItemInner)`
  background-image: url(${logo2});
  background-size: contain;
`;

const CloseButton = styled.div`
  z-index: 3;
  color: #fff;
  font-size: 25px;
  font-family: 'Orbitron', sans-serif;
  cursor: pointer;
  transition: all 0.2s ease-in;
  padding-left: 3px;
`;

const Setting = styled.div`
  z-index: 3;
  transition: all 0.2s ease-in;
  padding-right: 3px;
  a {
    color: #fff;
    font-size: 1.2rem;
    font-family: 'Orbitron', sans-serif;
    cursor: pointer;
  }
`;

const CoffeeCat = styled.div`
  width: ${(props) => (props.width ? props.width : '30%')};
  height: ${(props) => (props.height ? props.height : '20%')};
  background-image: url(${cofeeCat});
  background-repeat: no-repeat;
  background-size: contain;
`;

export {
    PokerContainer, PokerListWrap, PokerSelectWrap,
    PokerItemWrap, PokerItemMask, PokerItem, PokerItemInner,
    CloseButton, Setting,
    PokerSelectSafari, PokerSelectMask, PokerSelectContainer,
    PokerSelectedItemFront, PokerSelectedItemFrontWrap, PokerSelectedItemFrontMask,
    PokerSelectedItemBack, PokerSelectedItemBackWrap, PokerSelectedItemBackMask, PokerSelectedItemBackInner,
    CoffeeCat
};
