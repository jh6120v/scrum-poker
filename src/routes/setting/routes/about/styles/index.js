import styled from 'styled-components';

const AboutContent = styled.div`
  width: 100%;
  padding: 20px;
  font-family: Arial, serif;
  font-size: .8rem;
  line-height: 1.2rem;
`;

const AppName = styled.h1`
  width: 100%;
  text-align: center;
  font-size: 1.5rem;
  margin-bottom: 15px;
  font-weight: bold;
`;

export {
    AboutContent,
    AppName
};