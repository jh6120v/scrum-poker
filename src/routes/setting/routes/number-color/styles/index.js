import styled from 'styled-components';

const ColorWatch = styled.div`
  padding: 3px;
  background: #fff;
  border-radius: 1px;
  box-shadow: 0 0 0 1px rgba(0,0,0,.1);
  display: inline-block;
  cursor: pointer;
  position: relative;
`;

const ColorWatchShow = styled.div`
  width: 30px;
  height: 15px;
  border-radius: 2px;
  background-color: ${(props) => (props.color ? props.color : '#FFF')};
`;

const ColorPicker = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  margin-top: 15px;
`;

export {
    ColorWatch,
    ColorWatchShow,
    ColorPicker
};