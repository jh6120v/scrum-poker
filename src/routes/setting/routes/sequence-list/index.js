import SequenceList from './containers/sequence-list';

export default SequenceList;