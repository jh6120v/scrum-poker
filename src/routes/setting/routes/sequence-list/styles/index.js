import styled from 'styled-components';

const Sequence = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
`;

const Name = styled.div`
  width: 100%;
`;

const List = styled.div`
  width: 100%;
  font-size: .7rem;
  line-height: .8rem;
`;

export {
    Sequence,
    Name,
    List
};