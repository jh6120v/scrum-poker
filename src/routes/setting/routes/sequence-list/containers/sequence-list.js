import React, { useCallback, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import MdCheckmark from 'react-ionicons/lib/MdCheckmark';
import { SettingItem, SettingWrap } from '../../../styles';
import { headerTitleSet, prevLinkActGoBack } from '../../../../../modules/header';
import { personalSequenceTypeChange } from '../../../../../modules/personal-setting';
import { pointListDataSet } from '../../../../../modules/point-list';
import { Name, Sequence, List } from '../styles';
import { history } from '../../../../../store';

const SequenceList = () => {
    const dispatch = useDispatch();
    const sequenceList = useSelector((state) => state.sequence.sequenceList);
    const personal = useSelector((state) => state.personal);

    useEffect(() => {
        dispatch(headerTitleSet({
            title: 'Sequence Type'
        }));
        dispatch(prevLinkActGoBack());
    }, [dispatch]);

    const changeSequenceType = useCallback((type) => {
        dispatch(personalSequenceTypeChange(type));

        dispatch(pointListDataSet({
            listName: type,
            listData: sequenceList[type]
        }));

        history.goBack();
    }, [dispatch, sequenceList]);

    return (
        <SettingWrap>
            {
                Object.keys(sequenceList).map((val) => (
                    <SettingItem key={val} onClick={() => changeSequenceType(val)} alignItemsCenter>
                        <Sequence>
                            <Name>{val.replace('-', ' ')}</Name>
                            <List>
                                (
                                {
                                    sequenceList[val].reduce((accumulator, currentValue, currentIndex) => {
                                        if (currentIndex < 6) {
                                            accumulator.push(currentValue.point);
                                        }

                                        return accumulator;
                                    }, []).join(',')
                                }
                                {
                                    sequenceList[val].length > 5 ? ` ... ${sequenceList[val][sequenceList[val].length - 1].point}` : null
                                }
                                )
                            </List>
                        </Sequence>
                        {val === personal.sequenceType ? <MdCheckmark color="#4f6571" /> : null}
                    </SettingItem>
                ))
            }
        </SettingWrap>
    );
};

export default SequenceList;
