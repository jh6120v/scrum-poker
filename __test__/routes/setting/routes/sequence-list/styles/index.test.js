import 'jest-styled-components';
import { render } from '@testing-library/react';
import React from 'react';
import { List, Name, Sequence } from '../../../../../../src/routes/setting/routes/sequence-list/styles';

describe('test sequence-list styles.', () => {
    it('test <Sequence /> snapshot.', () => {
        const { container } = render(<Sequence />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Sequence /> when render it.', () => {
        const { container } = render(<Sequence />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '70%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
    });

    it('test <Name /> when render it.', () => {
        const { container } = render(<Name />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
    });

    it('test <List /> snapshot.', () => {
        const { container } = render(<List />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <List /> when render it.', () => {
        const { container } = render(<List />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('font-size', '.7rem');
        expect(tree).toHaveStyleRule('line-height', '.8rem');
    });
});