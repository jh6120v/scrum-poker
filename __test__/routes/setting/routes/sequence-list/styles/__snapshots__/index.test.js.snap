// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sequence-list styles. test <List /> snapshot. 1`] = `
.c0 {
  width: 100%;
  font-size: .7rem;
  line-height: .8rem;
}

<div
  class="c0"
/>
`;

exports[`test sequence-list styles. test <Sequence /> snapshot. 1`] = `
.c0 {
  width: 70%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
/>
`;
